homeassistant/components/rointe/sensor.py

```diff
@@ -0,0 +1,108 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .coordinator import RointeDataUpdateCoordinator, RointeSensorEntityDescription
+from .rointe_entity import RointeRadiatorEntity
+
+SENSOR_DESCRIPTIONS = [
+    # Current room temperature sensor (probe value).
+    RointeSensorEntityDescription(
+        key="current_temperature",
+        name="Current Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda radiator: radiator.temp_probe,
+        last_reset_fn=lambda radiator: None,
+    ),
+    # Energy usage in Kw/h.
+    RointeSensorEntityDescription(
+        key="energy",
+        name="Energy Consumption",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda radiator: radiator.energy_data.kwh
+        if radiator.energy_data
+        else None,
+        last_reset_fn=lambda radiator: radiator.energy_data.start
+        if radiator.energy_data
+        else None,
+    ),
+    # Effective power usage in W.
+    RointeSensorEntityDescription(
+        key="power",
+        name="Effective Power",
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=POWER_WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda radiator: radiator.energy_data.effective_power
+        if radiator.energy_data
+        else None,
+        last_reset_fn=lambda radiator: None,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the radiator sensors from the config entry."""
+    coordinator: RointeDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    coordinator.add_sensor_entities_for_seen_keys(
+        async_add_entities, SENSOR_DESCRIPTIONS, RointeGenericSensor
+    )
+
+
+class RointeGenericSensor(RointeRadiatorEntity, SensorEntity):
+    """Generic radiator sensor."""
+
+    entity_description: RointeSensorEntityDescription
+
+    def __init__(
+        self,
+        radiator: RointeDevice,
+        coordinator: RointeDataUpdateCoordinator,
+        description: RointeSensorEntityDescription,
+    ) -> None:
+        """Initialize a generic sensor."""
+        super().__init__(
+            coordinator,
+            radiator,
+            name=f"{radiator.name} {description.name}",
+            unique_id=f"{radiator.id}-{description.key}",
+        )
+
+        self.entity_description = description
```

Comment:
If you're assigning `self.entity_description`, you don't need to set `name` manually—it will automatically be pulled from the entity description.

https://github.com/home-assistant/core/blob/a2935654b9237ddc09cf5ea67c8bd967c660ce91/homeassistant/helpers/entity.py#L320-L327

Reply:
I changed this a bit, after removing the name all the sensor's entity names were changed to the format "sensor.current_temperature_2, _3, etc..."

I added a lambda to define a name based on the device's name:

`name_fn=lambda radiator: f"{radiator.name} Current Temperature",`
as for the Entity ID, they remain incrementally named.

Let me know if this is OK.